source/arm9/sound.c

```diff
@@ -47,12 +50,17 @@ int soundPlayPSG(DutyCycle cycle, u16 freq, u8 volume, u8 pan){
 	msg.SoundPsg.volume = volume;
 	msg.SoundPsg.pan = pan;
 
+	comutex_acquire(&sound_fifo_mutex);
+
 	fifoSendDatamsg(FIFO_SOUND, sizeof(msg), (u8*)&msg);
+	fifoWaitValueAsync32(FIFO_SOUND);
+	int result = fifoGetValue32(FIFO_SOUND);
 
-	while(!fifoCheckValue32(FIFO_SOUND));
+	comutex_acquire(&sound_fifo_mutex);
```

Comment:
Shouldn't this be `comutex_release`?

Reply:
Yeah, I've made the same mistake all over the file...